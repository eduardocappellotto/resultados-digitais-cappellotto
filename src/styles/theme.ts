const theme = {
    grid: {
        container: '128rem'
    },
    font: {
        regular: 400,
        bold: 700,
        extrabold: 800,
        sizes: {
            xs: '1.2rem',
            sm: '1.4rem',
            md: '1.6rem',
            lg: '2rem',
            xlg: '2.4rem',
            '2xlg': '4rem'
        }
    },
    colors: {
        primary: {
            200: '#D8D9FD',
            500: '#6C5AE0',
            700: '#4839A3',
            900: '#100C27'
        },
        secondary: {
            200: '#F8DCA0',
            400: '#F4CA6D',
            500: '#F2BF4E'
        },
        gray: {
            200: '#E1E4E8',
            300: '#CFD3D8',
            400: '#97A1AC',
            500: '#636E7C',
            600: '#7E8A98'
        },
        white: '#FFFFFF',
        black: '#000000',
        light: '#F1F3F5',
        error: '#f87171',
        dark: '#212429',

        gradient: 'linear-gradient(93.82deg, #6C5AE0 -1.6%, #806EFF 105.56%)',

    },
    spacings: {
        xs: '1.6rem',
        sm: '2.4rem',
        md: '3.2rem',
        lg: '4.8rem'
    }
}

export default theme
