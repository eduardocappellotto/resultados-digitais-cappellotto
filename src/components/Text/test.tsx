import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { renderWithTheme } from "@/utils/tests/helpers";
import { Text, TextProps } from ".";
import theme from "@/styles/theme";

describe("<Text />", () => {
  const renderText = (props: TextProps) => {
    return renderWithTheme(<Text {...props} />);
  };

  it("should render the white text as default", () => {
    renderText({ children: "Random text" });

    expect(screen.getByText(/random text/i)).toHaveStyle({
      color: theme.colors.white,
    });
    expect(screen.getByText(/random text/i)).toHaveStyle({
      "font-size": theme.font.sizes.xs,
    });
  });

  it("should render the text with different sizes", () => {
    renderText({ children: "First text", size: "medium" });
    renderText({ children: "Second text", size: "large" });

    expect(screen.getByText(/first text/i)).toHaveStyle({
      "font-size": theme.font.sizes.sm,
    });
    expect(screen.getByText(/second text/i)).toHaveStyleRule(
      "font-size",
      theme.font.sizes.lg,
      {
        media: "(min-width:  768px)",
      }
    );
  });

  it("should render as different HTML elements when 'as' prop is provided", () => {
    renderText({ children: "Span Text", as: "span" });
    renderText({ children: "Heading Text", as: "h2" });

    expect(screen.getByText(/span text/i).tagName).toBe("SPAN");
    expect(screen.getByText(/heading text/i).tagName).toBe("H2");
  });

  it("should render text with different colors when 'color' prop is provided", () => {
    renderText({ children: "Black Text", color: "black" });
    renderText({ children: "Dark Text", color: "dark" });

    expect(screen.getByText(/black text/i)).toHaveStyle({
      color: theme.colors.black,
    });
    expect(screen.getByText(/dark text/i)).toHaveStyle({
      color: theme.colors.dark,
    });
  });

  it("should render text with different font weights when 'weight' prop is provided", () => {
    renderText({ children: "Bold Text", weight: "bold" });
    renderText({ children: "Regular Text", weight: "regular" });

    expect(screen.getByText(/bold text/i)).toHaveStyle({
      "font-weight": theme.font.bold.toString(),
    });
    expect(screen.getByText(/regular text/i)).toHaveStyle({
      "font-weight": theme.font.regular.toString(),
    });
  });
});
