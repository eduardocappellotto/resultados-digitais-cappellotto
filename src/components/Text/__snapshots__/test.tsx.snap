// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match snapshots for different prop combinations 1`] = `
.c0 {
  color: #212429;
  line-height: 1.7;
  font-weight: 400;
  font-size: 1.4rem;
}

@media (min-width: 768px) {

}

<div>
  <p
    class="c0"
  >
    Medium Dark Text
  </p>
</div>
`;

exports[`<Text /> should match snapshots for different prop combinations 2`] = `
.c0 {
  color: #FFFFFF;
  line-height: 1.7;
  font-weight: 700;
  font-size: 1.4rem;
}

@media (min-width: 768px) {
  .c0 {
    font-size: 2rem;
  }
}

<div>
  <p
    class="c0"
  >
    Large Bold Text
  </p>
</div>
`;

exports[`<Text /> should render the white text as default 1`] = `
.c0 {
  color: #FFFFFF;
  line-height: 1.7;
  font-weight: 400;
  font-size: 1.2rem;
}

<p
  class="c0"
>
  Random text
</p>
`;
